packages/core/src/app/payment/Payment.tsx

```diff
@@ -519,7 +520,9 @@ export function mapToPaymentProps({
     const customer = getCustomer();
     const consignments = getConsignments();
     const { isComplete = false } = getOrder() || {};
-    const methods = getPaymentMethods() || EMPTY_ARRAY;
+    const stripeUpePaymentMethod = getPaymentMethod('card', PaymentMethodId.StripeUPE);
+    const methods = props.isStripeLinkAuthenticated && stripeUpePaymentMethod ?
```

Comment:
Should this happen on payments endpoint? Or we want to do this in frontend?

Reply:
hey @animesh1987  We do it in the front because in the back we don't have the way to know when the shopper is authenticated with his Link account

Reply:
Sorry I haven't had time to test the paypal flow, but as far as I know if we click paypal button on cart page. We filter other providers on checkout payment step. 
@serhii-tkachenko do you know how that logic is setup?

Reply:
As far as I know, provider setup step triggers on bcapp side.
When customer submits paypal flow we are posting the form through formPoster here:
https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts#L87

Then, on the backend side we are selecting specific provider (and returning only paypal in array of payment providers for payment step). You can double check this code on bcapp:
https://github.com/bigcommerce/bigcommerce/blob/01f0b9e7b92e3cdf75cd03bda50d838d1de3a219/modules/checkout/paypalcommerce/module.paypalcommerce.php#L727

Reply:
Hey @animesh1987 , we tried to do different things to support this in the backend but we’re detecting multiple issues with this approach:
	1. Currently the use of setRemoteProvider (through set_external_checkout) as other payment providers are using, is not prepared for MultiOption Providers, currently, this flow only accepts the gateway, and the APM it's not.
	2. Now, Stripe Link doesn't support the authentication after a page reload (because it is not preserved on the stripe side, that part is in their scope but not worked yet). So checking the flow that PayPal is following, requires a page reload to load the remote provider.
	3. To support remoteProvider for MultiOption requires a bigger effort because it touches multiple repos (Bcapp, Checkout-SDK & Bigpay) and before we start to code something we need to discuss together with the payments team because apparently, this approach touches core files.
Saying this, we are in a very complicated position because the timeline is super tight and there is a contractual date and financial agreement that is a big pressure for us to deliver this. So I propose the following actions:
	1. Send the code that works good (adding a validation into Payment.tsx) and add a TODO comment that we're gonna change that approach to get the provider by the backend after the certification and as fast follow, also adding the ticket number (this ticket will be created if you agree of this :pray)  where it will be addressed.
    2. Create a thread to discuss these problematics related to RemoteProvider and the flow that we're trying to implement with Stripe Link, to include multiple BC teams that maybe can give us a better context and ideas to solve the problem/concern you're describing and after the certification from Stripe.

I hope you understand the complex situation we have related to the deadline (Oct 14th in Prod).

Let me know if you have any other questions about it, 

CC: @clopezh, @vmparra, @FranciscoOnt